doc

justinmk opened the pull request:
- [x] remove textprop references https://github.com/neovim/neovim/commit/aa680f6acb84b876584cbe5c958d50c692154c75#r34755682

justinmk committed:
doc

justinmk committed:
doc: eliminate msgpack_rpc.txt [ci skip]

- Migrate msgpack_rpc.txt into api.txt, develop.txt.
- fix #10740: Remove warning about "avoid hardcoding the type codes".

justinmk committed:
doc: |api-fast| [ci skip]

justinmk commented:
@blueyed thanks. JFI that doc is generated from the C sources :)

zgpio commented:
There seems to be a redundant space here.
https://github.com/neovim/neovim/blob/6a21fe9fd16cfedc87b58064d09218e5c06ddd89/runtime/doc/api.txt#L715
https://github.com/neovim/neovim/blob/6a21fe9fd16cfedc87b58064d09218e5c06ddd89/runtime/doc/api.txt#L727

justinmk commented:
> There seems to be a redundant space here.

It's not easy to fix. `scripts/gen_vimdoc.py` has to make some ambiguous decisions, the space is a tradeoff. And a pretty harmless one.

bfredl committed:
doc/API/lua: detaching Lua buffer callbacks

justinmk commented on src/nvim/api/buffer.c:
```diff
@@ -181,7 +181,8 @@ Boolean nvim_buf_attach(uint64_t channel_id,
 
-/// Deactivates buffer-update events on the channel.
+/// Deactivates buffer-update events on the channel. Not usable from lua,
+/// for attached lua callbacks see |api-buffer-updates-lua-detach|.
```
not in VimL either. I guess this could be in the Attributes section of the docs.

zgpio commented:
Extra blank line
https://github.com/neovim/neovim/blob/1d59575450e0a09d710a7cad2b8e1b5c5e0242f5/runtime/doc/api.txt#L817-L819
https://github.com/neovim/neovim/blob/1d59575450e0a09d710a7cad2b8e1b5c5e0242f5/runtime/doc/api.txt#L1116-L1118
Affected the outline:
![image](https://user-images.githubusercontent.com/19503791/63856092-72a1ce80-c9d3-11e9-9a43-702a45873cdc.png)

zgpio commented:
It seems  `or 0 for current window` is missing.
https://github.com/neovim/neovim/blob/1d59575450e0a09d710a7cad2b8e1b5c5e0242f5/runtime/doc/api.txt#L1746-L1751

blueyed commented:
Needs to be rebased (conflicts).
(wanted to push a minor fix for "wait()" (indent))

justinmk committed:
doc: StatusLineTerm, StatusLineTermNC

N/A:
vim-patch:8.0.0941: existing color schemes don't like StatusLineTerm
vim-patch:8.0.0937: user highlight groups not adjusted for terminal
vim-patch:8.0.0825: not easy to see that a window is a terminal window

justinmk commented:
Still more doc work needed, but merging this for now.

> Extra blank line ... Affected the outline:

@zgpio that will take time to fix (need to debug `gen_vimdoc.py`).
